### PR TITLE
Remove potential deadlock

### PR DESCRIPTION
**Pull request by lugray**
If someone took the lock to `Broadcast`, then someone started trying to `cancelWait`, the `select` went to the `default` because no one was listening on `c.cancel`, then `Broadcast` was called, `Broadcast` would have been trying to send to `c.signal` with no one listening, and `cancelWait` would never get it's lock.

### CODE REVIEW

File: cond/cond.go [hunk 1]
```diff
@@ -88,15 +88,27 @@ func (c *Cond) ContextWait(ctx context.Context) bool {
 }
 
 func (c *Cond) cancelWait() {
+	gotLock := make(chan struct{})
+	cancelled := false
+
+	go func() {
+		c.L.Lock()
+		if cancelled {
```
Comment (lavoiesl) [inline cond/cond.go]: There’s a data race on `cancelled`.

It’s possible for the Signal to receive the cancel and for the client to unlock, both before it was able to set the `cancelled` boolean.

I added a test to your branch that shows this.

Comment (lugray) [reply cond/cond.go]: Nice catch.  Thanks, fixed.

Comment (lavoiesl) [reply cond/cond.go]: Ah! Good job! I was trying to make it work, but was failing miserably.
I fixed my test, it's not simpler and more meaningful.
I forced-pushed to your branch.

**Comment by lugray** [comment]
@lavoiesl You need to set up automatic gofmt on save. 😑

**Pull request merged**